### PR TITLE
Do not load components from Storage on Buffer API start

### PR DESCRIPTION
**Pull request by JakubMatejka**
Takhle nějak? 😅

### CODE REVIEW

File: internal/pkg/service/buffer/dependencies/dependencies.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func NewServiceDeps(
 	// Create public dependencies - load API index
 	startTime := time.Now()
 	logger.Info("loading Storage API index")
-	publicDeps, err := dependencies.NewPublicDeps(ctx, baseDeps, storageAPIHost)
+	publicDeps, err := dependencies.NewPublicDeps(ctx, baseDeps, storageAPIHost, false)
```
Comment (michaljurecko) [inline internal/pkg/service/buffer/dependencies/dependencies.go:72]: Mohlo by byt z toho prosim `WithComponents(false)`? 
Ono sa to lepsie cita a vies co to robi, ... a nechcem tam pridavat asi dalsie `bool flags`.

Comment (JakubMatejka) [reply internal/pkg/service/buffer/dependencies/dependencies.go:72]: aha, ok

Comment (michaljurecko) [reply internal/pkg/service/buffer/dependencies/dependencies.go:72]: example https://github.com/keboola/keboola-as-code/blob/main/internal/pkg/service/buffer/worker/distribution/config.go

File: internal/pkg/service/cli/dependencies/local.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ func newPublicDeps(baseDeps Base) (*local, error) {
 	}
 
 	// Create common local dependencies
-	publicDep, err := dependencies.NewPublicDeps(baseDeps.CommandCtx(), baseDeps, host, true)
+	publicDep, err := dependencies.NewPublicDeps(baseDeps.CommandCtx(), baseDeps, host, dependencies.WithComponents(true))
```
Comment (michaljurecko) [inline internal/pkg/service/cli/dependencies/local.go:53]: Default value by som dal `true` a teda to tu byt nemusi (alebo false ak ti to dava vacsiu logiku).
Pri `default = false` by sa to mohlo volat `WithPreloadComponents()`.

File: internal/pkg/service/common/dependencies/public.go [hunk 3]
```diff
@@ -24,13 +24,32 @@ type public struct {
 	components          Lazy[*model.ComponentsProvider]
 }
 
-func NewPublicDeps(ctx context.Context, base Base, storageAPIHost string, loadComponents bool) (v Public, err error) {
+type PublicDepsOption func(*publicDepsConfig)
+
+type publicDepsConfig struct {
+	includeComponents bool
```
Comment (michaljurecko) [inline internal/pkg/service/common/dependencies/public.go:30]: mozno by bolo jasnejsie `preloadComponents`?

**Review by michaljurecko — APPROVED** [review]
LGTM

File: internal/pkg/service/cli/dependencies/local.go (hunk 2 above)
Comment (michaljurecko) [reply internal/pkg/service/cli/dependencies/local.go:53]: Ako to po sebe citam, tak to asi napisem podrobnejsie :)

Comment (michaljurecko) [reply internal/pkg/service/cli/dependencies/local.go:53]: Teraz, ked nic nezadas, tak je `includeComponents=false`, teda nerobi `preload`.

1. To je myslim, celkom logicke, len nevidim dovod potom davat na dane miesta `WithComponents`.

2. A ako nad tym rozmyslam, tak mozno by bolo este lepsie `WithPreloadComponents()` (bez argumentu), alebo ``WithPreloadComponents(bool)`.

Ale to je uz na tebe, ci ti to dava zmysel alebo nie.

File: internal/pkg/service/common/dependencies/public.go [hunk 4]
```diff
@@ -21,32 +21,61 @@ type public struct {
...
 }
 
-func NewPublicDeps(ctx context.Context, base Base, storageAPIHost string) (v Public, err error) {
+type PublicDepsOption func(*publicDepsConfig)
+
+type publicDepsConfig struct {
+	includeComponents bool
+}
+
+// WithComponents defines if the components should be retrieved from Storage index.
```
Comment (michaljurecko) [inline internal/pkg/service/common/dependencies/public.go:33]: Tento komentar by sa mohol mozno este vylepsit, `be retrieved from Storage index ON STARTUP`/`ON INIT`?

**Review by michaljurecko — APPROVED** [review]
:+1:

**Pull request merged**